openshift-client/src/test/java/io/fabric8/openshift/client/mock/RoleBindingTest.java

```diff
@@ -0,0 +1,249 @@
+
+    // Initial GET
+    server.getMockServer().takeRequest();
+    RecordedRequest request = server.getMockServer().takeRequest();
+    assertEquals(expectedRoleBinding, new ObjectMapper().readerFor(RoleBinding.class).readValue(request.getBody().inputStream()));
+  }
+
+  @Test
+  public void testReplaceWithUserNamesAndGroupsAndNoSubjects() throws Exception {
+    server.expect().get().withPath("/oapi/v1/namespaces/testns/rolebindings/testrb").andReturn(200, expectedRoleBinding).once();
+    server.expect().put().withPath("/oapi/v1/namespaces/testns/rolebindings/testrb").andReturn(200, expectedRoleBinding).once();
+
+    NamespacedOpenShiftClient client = server.getOpenshiftClient();
+
+    RoleBinding response = client.inNamespace("testns").roleBindings().withName("testrb").replace(
+      new RoleBindingBuilder()
+        .withNewMetadata().endMetadata()
+        .addToUserNames("testuser1", "testuser2", "system:serviceaccount:testns:svcacct")
+        .addToGroupNames("testgroup")
+        .build()
+    );
+    assertEquals(expectedRoleBinding, response);
+
+    // Initial GET
+    server.getMockServer().takeRequest();
+    RecordedRequest request = server.getMockServer().takeRequest();
+    assertEquals(expectedRoleBinding, new ObjectMapper().readerFor(RoleBinding.class).readValue(request.getBody().inputStream()));
+  }
+
+  @Test
+  public void testReplaceWithUserNamesAndGroupsAndOverwriteSubjects() throws Exception {
+    server.expect().get().withPath("/oapi/v1/namespaces/testns/rolebindings/testrb").andReturn(200, expectedRoleBinding).once();
+    server.expect().put().withPath("/oapi/v1/namespaces/testns/rolebindings/testrb").andReturn(200, expectedRoleBinding).once();
+
+    NamespacedOpenShiftClient client = server.getOpenshiftClient();
+
+    RoleBinding response = client.inNamespace("testns").roleBindings().withName("testrb").replace(
+      new RoleBindingBuilder()
+        .withNewMetadata().endMetadata()
+        .addToUserNames("testuser1", "testuser2", "system:serviceaccount:testns:svcacct")
+        .addToGroupNames("testgroup")
+        .addNewSubject().withKind("User").withName("unexpected").endSubject()
+      .build()
+    );
+    assertEquals(expectedRoleBinding, response);
+
+    // Initial GET
+    server.getMockServer().takeRequest();
+    RecordedRequest request = server.getMockServer().takeRequest();
+    assertEquals(expectedRoleBinding, new ObjectMapper().readerFor(RoleBinding.class).readValue(request.getBody().inputStream()));
+  }
+
+  @Test
+  public void testPatchWithOnlySubjects() throws Exception {
+    server.expect().get().withPath("/oapi/v1/namespaces/testns/rolebindings/testrb").andReturn(200, new RoleBindingBuilder().addToUserNames("unexpected").build()).once();
+    server.expect().patch().withPath("/oapi/v1/namespaces/testns/rolebindings/testrb").andReturn(200, expectedRoleBinding).once();
+
+    NamespacedOpenShiftClient client = server.getOpenshiftClient();
```

Comment:
Why use a NamespacedOpenshiftClient if you are going to use inNamespace("testns") afterwards?